File: backend_node_backup/verifyConnection.js
import mysql from 'mysql2/promise';
import dotenv from 'dotenv';

dotenv.config();

const pool = mysql.createPool({
  host: process.env.DB_HOST || 'localhost',
  port: process.env.DB_PORT || 3306,
  user: process.env.DB_USER || 'root',
  password: process.env.DB_PASSWORD || '',
  database: process.env.DB_NAME || 'mitienda_db',
  waitForConnections: true,
  connectionLimit: 5,
  queueLimit: 0,
});

console.log('========================================================');
console.log('🔌  PRUEBA DE CONEXIÓN: BACKEND NODE.JS + EXPRESS');
console.log('========================================================');
console.log();

let conn;
try {
  conn = await pool.getConnection();
  console.log('✅  Conexión exitosa a MySQL');
  console.log();

  const [now] = await conn.query('SELECT NOW() AS fecha_hora');
  console.log(`🕒  Fecha/hora del servidor: ${now[0].fecha_hora}`);

  const [version] = await conn.query('SELECT VERSION() AS v');
  console.log(`🗄️  Motor: MySQL/MariaDB ${version[0].v}`);
  console.log();

  console.log('📋  Verificación de tablas y datos:');
  const checks = [
    ['roles', 'SELECT COUNT(*) FROM roles'],
    ['permisos', 'SELECT COUNT(*) FROM permisos'],
    ['rol_permisos', 'SELECT COUNT(*) FROM rol_permisos'],
    ['usuarios', 'SELECT COUNT(*) FROM usuarios'],
    ['productos', 'SELECT COUNT(*) FROM productos'],
    ['servicios', 'SELECT COUNT(*) FROM servicios'],
    ['codigos_recuperacion', 'SELECT COUNT(*) FROM codigos_recuperacion'],
  ];

  for (const [tabla, sql] of checks) {
    const [r] = await conn.query(sql);
    const cnt = Object.values(r[0])[0];
    const estado = cnt > 0 ? '✅' : '⚠️ ';
    console.log(`   ${estado} ${tabla.padEnd(22)} ${cnt} registros`);
  }
  console.log();

  console.log('🔐  Verificación de credenciales admin (hash):');
  const [adm] = await conn.query("SELECT id, email, rol_id, LEFT(password, 20) AS pass_preview FROM usuarios WHERE email = 'admin@mitienda.com'");
  if (adm.length > 0) {
    console.log(`   ✅  admin@mitienda.com encontrado (id=${adm[0].id}, rol_id=${adm[0].rol_id})`);
    console.log(`       Hash bcrypt: ${adm[0].pass_preview}...`);
  } else {
    console.log('   ❌  Admin NO encontrado');
  }
  console.log();

  console.log('🛍️  Productos disponibles:');
  const [prods] = await conn.query('SELECT id, nombre, precio, stock FROM productos ORDER BY id');
  for (const p of prods) {
    const precio = new Intl.NumberFormat('es-CO', { style: 'currency', currency: 'COP', maximumFractionDigits: 0 }).format(Number(p.precio));
    console.log(`   [${p.id}] ${p.nombre.padEnd(28)} ${precio.padStart(14)} | Stock: ${p.stock}`);
  }
  console.log();

  console.log('🔧  Servicios disponibles (con empleado asignado):');
  const [servs] = await conn.query(`
    SELECT s.id, s.nombre, s.precio, s.duracion,
           CONCAT(u.nombre, ' ', u.apellido) AS empleado
    FROM servicios s
    LEFT JOIN usuarios u ON u.id = s.usuario_id
    ORDER BY s.id
  `);
  for (const s of servs) {
    const precio = new Intl.NumberFormat('es-CO', { style: 'currency', currency: 'COP', maximumFractionDigits: 0 }).format(Number(s.precio));
    console.log(`   [${s.id}] ${s.nombre.padEnd(26)} ${precio.padStart(14)} | ${s.duracion.padEnd(10)} | ${s.empleado || 'Sin asignar'}`);
  }
  console.log();

  console.log('🎉  BACKEND NODE.JS: CONEXIÓN 100% FUNCIONAL');
  console.log('   Puedes arrancar con: npm run dev   (puerto 3000)');

} catch (e) {
  console.error('❌  ERROR DE CONEXIÓN:', e.message);
  process.exitCode = 1;
} finally {
  if (conn) conn.release();
  await pool.end();
}

File: Backend/verifyConnection.py
import sys
import os

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

if sys.stdout.encoding != 'utf-8':
    try:
        sys.stdout.reconfigure(encoding='utf-8')
    except Exception:
        pass

from dotenv import load_dotenv
import pymysql

load_dotenv()

DB_HOST = os.getenv('DB_HOST', 'localhost')
DB_PORT = int(os.getenv('DB_PORT', '3306'))
DB_USER = os.getenv('DB_USER', 'root')
DB_PASSWORD = os.getenv('DB_PASSWORD', '')
DB_NAME = os.getenv('DB_NAME', 'mitienda_db')

print("=" * 56)
print("🔌  PRUEBA DE CONEXIÓN: BACKEND FASTAPI + PYTHON")
print("=" * 56)
print()

try:
    conn = pymysql.connect(
        host=DB_HOST, port=DB_PORT, user=DB_USER,
        password=DB_PASSWORD, database=DB_NAME,
        charset='utf8mb4', cursorclass=pymysql.cursors.DictCursor
    )
    print("✅  Conexión exitosa a MySQL (pymysql)")
    print()

    with conn.cursor() as cur:
        cur.execute("SELECT NOW() AS fh;")
        print(f"🕒  Fecha/hora: {cur.fetchone()['fh']}")

        cur.execute("SELECT VERSION() AS v;")
        print(f"🗄️  Motor: {cur.fetchone()['v']}")
    print()

    print("📋  Verificación de tablas y datos:")
    checks = [
        'roles', 'permisos', 'rol_permisos', 'usuarios',
        'productos', 'servicios', 'codigos_recuperacion',
        'pedidos', 'pedido_items', 'citas_servicios'
    ]
    with conn.cursor() as cur:
        for t in checks:
            cur.execute(f"SELECT COUNT(*) AS c FROM {t};")
            cnt = cur.fetchone()['c']
            estado = "✅" if cnt > 0 or t in ('codigos_recuperacion', 'pedidos', 'pedido_items', 'citas_servicios') else "⚠️ "
            print(f"   {estado} {t.ljust(22)} {cnt} registros")
    print()

    print("🔐  Admin user:")
    with conn.cursor() as cur:
        cur.execute("SELECT id, email, rol_id, estado FROM usuarios WHERE email='admin@mitienda.com';")
        a = cur.fetchone()
        if a:
            print(f"   ✅  Encontrado -> id={a['id']} rol={a['rol_id']} estado={a['estado']}")
        else:
            print("   ❌  NO ENCONTRADO")
    print()

    print("🔗  Prueba JOIN (servicio -> empleado):")
    with conn.cursor() as cur:
        cur.execute("""
            SELECT s.nombre AS servicio,
                   CONCAT(u.nombre, ' ', u.apellido) AS empleado
            FROM servicios s
            JOIN usuarios u ON u.id = s.usuario_id
            ORDER BY s.id;
        """)
        for r in cur.fetchall():
            print(f"   - {r['servicio'].ljust(26)} -> {r['empleado']}")
    print()

    print("🎉  BACKEND FASTAPI: CONEXIÓN 100% FUNCIONAL")
    print("   (sqlalchemy requiere pip install -r requirements.txt)")
    print("   Puedes arrancar con: python run.py  (puerto 8000)")

    conn.close()

except Exception as e:
    print(f"❌  ERROR: {e}")
    import traceback
    traceback.print_exc()
    sys.exit(1)
